packages/core-sdk/resolvers/token-resolvers/resolvers/aave/src/query/functions/getTokenComponents.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  env,
+  Ethereum_Connection,
+  Ethereum_Query,
+  Input_getTokenComponents,
+  Interface_Token,
+  Interface_TokenComponent,
+  QueryEnv,
+  Token_Query,
+  Token_Token,
+  Token_TokenType,
+} from "../w3";
+
+function fetchUnderlyingTokenAddress(token: Token_Token, connection: Ethereum_Connection): string {
```

Comment:
Looking at this piece of code, I realized it would be better to also take the protocol version as input of`getTokenComponents` to avoid try...catching for versions in the token resolver.

Reply:
Cool, I'll add it in